packages/firestore/src/core/firestore_client.ts

```diff
@@ -227,40 +227,46 @@ export class FirestoreClient {
    *     continue, i.e. that one of the persistence implementations actually
    *     succeeded.
    */
-  private initializePersistence(
+  private async initializePersistence(
+    persistenceProvider: PersistenceProvider,
     persistenceSettings: InternalPersistenceSettings,
-    persistenceResult: Deferred<void>,
-    user: User
-  ): Promise<LruGarbageCollector | null> {
-    if (persistenceSettings instanceof IndexedDbPersistenceSettings) {
-      return this.startIndexedDbPersistence(user, persistenceSettings)
-        .then(maybeLruGc => {
-          persistenceResult.resolve();
-          return maybeLruGc;
-        })
-        .catch(error => {
-          // Regardless of whether or not the retry succeeds, from an user
-          // perspective, offline persistence has failed.
-          persistenceResult.reject(error);
-
-          // An unknown failure on the first stage shuts everything down.
-          if (!this.canFallback(error)) {
-            throw error;
-          }
-          console.warn(
-            'Error enabling offline persistence. Falling back to' +
-              ' persistence disabled: ' +
-              error
-          );
-          return this.startMemoryPersistence();
-        });
-    } else {
-      // When usePersistence == false, enabling offline persistence is defined
-      // to unconditionally succeed. This allows start() to have the same
-      // signature for both cases, despite the fact that the returned promise
-      // is only used in the enablePersistence call.
+    user: User,
+    persistenceResult: Deferred<void>
+  ): Promise<void> {
+    try {
+      await persistenceProvider.configure(
```

Comment:
Since persistence provider's `configure` method is called during `initializePersistence`, consider calling this method `initialize`.

Reply:
Done